apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/preadj/FissClaimTransformerV2.java

```diff
@@ -61,20 +90,40 @@ static Claim transform(MetricRegistry metricRegistry, Object claimEntity) {
   private static Claim transformClaim(PreAdjFissClaim claimGroup) {
     Claim claim = new Claim();
 
+    boolean isIcd9 =
+        claimGroup.getStmtCovToDate() != null
+            && claimGroup.getStmtCovToDate().isBefore(ICD_9_CUTOFF_DATE);
+
     claim.setId("f-" + claimGroup.getDcn());
-    claim.setMeta(new Meta().setLastUpdated(Date.from(claimGroup.getLastUpdated())));
-    claim.setContained(getContainedProvider(claimGroup));
+    claim.setContained(List.of(getContainedPatient(claimGroup), getContainedProvider(claimGroup)));
     claim.setIdentifier(getIdentifier(claimGroup));
+    claim.setExtension(getExtension(claimGroup));
     claim.setStatus(Claim.ClaimStatus.ACTIVE);
     claim.setType(getType());
+    claim.setSupportingInfo(getSupportingInfo(claimGroup));
+    claim.setBillablePeriod(getBillablePeriod(claimGroup));
     claim.setUse(Claim.Use.CLAIM);
-    claim.setCreated(new Date());
     claim.setPriority(getPriority());
     claim.setTotal(getTotal(claimGroup));
     claim.setProvider(new Reference("#provider-org"));
-    claim.setPatient(getPatient(claimGroup));
-    claim.setDiagnosis(getDiagnosis(claimGroup));
-    claim.setProcedure(getProcedure(claimGroup));
+    claim.setPatient(new Reference("#patient"));
+    claim.setFacility(getFacility(claimGroup));
+    claim.setDiagnosis(getDiagnosis(claimGroup, isIcd9));
+    claim.setProcedure(getProcedure(claimGroup, isIcd9));
+    claim.setInsurance(getInsurance(claimGroup));
+
+    FhirContext ctx = FhirContext.forR4();
+    IParser parser = ctx.newJsonParser();
+
+    String claimContent = parser.encodeResourceToString(claim);
+
+    String resourceHash = DigestUtils.sha1Hex(claimContent);
+
+    claim.setMeta(
+        new Meta()
+            .setLastUpdated(Date.from(claimGroup.getLastUpdated()))
+            .setVersionId("f-" + claimGroup.getDcn() + "-" + resourceHash));
```

Comment:
❓ Is this version id creation something that is used outside this one spot? Seems like something that could be extracted to a util if we expect this formatting will be done elsewhere.

Reply:
This is the only place the generated value is used to my knowledge, it is basically a checksum, and we don't keep track of it.